### DISCUSSION TITLE
ebpf 探测 go 应用的 interface 变量

### DISCUSSION THREAD

**Discussion by giscus** [edony-ink/comments#5] [General]
# ebpf 探测 go 应用的 interface 变量

interface 是 go 语言中最成功的设计之一，tracing interface 最大的技术难点是理解 go interface 的底层实现特别是运行时的内存布局，对此我打算从两个角度来结合理解：1.语法层面理解即从 go 源码角度理解 interface 是怎么实现；2.runtime 层面理解即借助 bpf uprobe 技术，从 go 应用运行时角度理解 interface 是怎么实现

https://www.edony.ink/understand-go-interface-with-ebpf-tech/

**Comment by edonyzpc** [comment]
从汇编等底层技术出发，详细地分析了 go interface 的底层实现和内存布局 👍
